File: scripts/sync-app-icons.py
#!/usr/bin/env python3
"""Convert brand/app-icons/{app} masters into apps/{app}/assets/app-icon.png (1024×1024 PNG)."""
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    raise SystemExit("Install Pillow: pip install Pillow")

ROOT = Path(__file__).resolve().parents[1]
ICONS_DIR = ROOT / "brand" / "app-icons"
APPS = ("farmer", "warehouse", "agronomist", "telecaller")
EXTENSIONS = (".jpeg", ".jpg", ".png", ".webp")
SIZE = 1024


def resolve_source(app: str) -> Path:
    for ext in EXTENSIONS:
        candidate = ICONS_DIR / f"{app}{ext}"
        if candidate.is_file():
            return candidate
    raise SystemExit(f"Missing source for {app} in {ICONS_DIR}")


def main() -> None:
    if not ICONS_DIR.is_dir():
        raise SystemExit(f"Missing icon directory: {ICONS_DIR}")

    for app in APPS:
        source = resolve_source(app)
        img = Image.open(source).convert("RGBA")
        if img.size != (SIZE, SIZE):
            img = img.resize((SIZE, SIZE), Image.Resampling.LANCZOS)

        dest_dir = ROOT / "apps" / app / "assets"
        dest_dir.mkdir(parents=True, exist_ok=True)
        dest = dest_dir / "app-icon.png"
        img.save(dest, format="PNG", optimize=True)
        print(f"Wrote {dest.relative_to(ROOT)} ← {source.relative_to(ROOT)}")


if __name__ == "__main__":
    main()

File: scripts/sync-app-icons.mjs
#!/usr/bin/env node
/**
 * Convert brand/app-icons/{app} masters into apps/{app}/assets/app-icon.png (1024×1024 PNG).
 * Used locally and on EAS via eas-build-pre-install.
 */
import { access, mkdir, readdir } from 'node:fs/promises';
import path from 'node:path';
import { fileURLToPath } from 'node:url';
import sharp from 'sharp';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ROOT = path.resolve(__dirname, '..');
const ICONS_DIR = path.join(ROOT, 'brand', 'app-icons');
const APPS = ['farmer', 'warehouse', 'agronomist', 'telecaller', 'partner'];
const SIZE = 1024;
const EXTENSIONS = ['.jpeg', '.jpg', '.png', '.webp'];

async function resolveSource(app) {
  for (const ext of EXTENSIONS) {
    const candidate = path.join(ICONS_DIR, `${app}${ext}`);
    try {
      await access(candidate);
      return candidate;
    } catch {
      // try next extension
    }
  }
  return null;
}

async function main() {
  try {
    await access(ICONS_DIR);
  } catch {
    console.error('Missing icon directory: brand/app-icons/');
    console.error('Add source files: brand/app-icons/farmer.jpeg, warehouse.jpeg, agronomist.jpeg, telecaller.jpeg');
    process.exit(1);
  }

  const available = await readdir(ICONS_DIR);
  console.log(`Sources in brand/app-icons/: ${available.join(', ')}`);

  for (const app of APPS) {
    const source = await resolveSource(app);
    if (!source) {
      console.error(`Missing source for ${app} (expected brand/app-icons/${app}.jpeg or .png)`);
      process.exit(1);
    }

    const destDir = path.join(ROOT, 'apps', app, 'assets');
    await mkdir(destDir, { recursive: true });
    const dest = path.join(destDir, 'app-icon.png');

    await sharp(source)
      .resize(SIZE, SIZE, { fit: 'cover' })
      .png({ compressionLevel: 9 })
      .toFile(dest);

    console.log(`Wrote ${path.relative(ROOT, dest)} ← ${path.relative(ROOT, source)}`);
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
